src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -75,6 +75,7 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, PageRouteMode
             {
                 var descriptor = new PageActionDescriptor
                 {
+                    ActionConstraints = selector.ActionConstraints,
```

Comment:
Make sense. However, this initialization also doesn't set at least the `EndpointMetadata`, `BoundProperties` or `Parameters` properties. Should it?

Reply:
It was missing `EndpointMetadata`. We don't calculate `BoundProperties` or `Parameters` until a request comes in and we've compiled the page.